packages/client/tests/functional/issues/8832/tests.ts

```diff
@@ -0,0 +1,192 @@
+      await prisma.tag.createMany({
+        data,
+      })
+      return ids
+    }
+
+    async function getTagsFindManyIn(ids: number[]): Promise<unknown[]> {
+      const tags = await prisma.tag.findMany({
+        where: {
+          id: {
+            in: ids,
+          },
+        },
+      })
+      return tags
+    }
+
+    async function getTagsFindManyInclude(): Promise<unknown[]> {
+      const tags = await prisma.tag.findMany({
+        include: {
+          posts: true,
+        },
+      })
+      return tags
+    }
+
+    async function clean() {
+      const cleanPrismaPromises = [prisma.tagsOnPosts.deleteMany(), prisma.post.deleteMany(), prisma.tag.deleteMany()]
+      await prisma.$transaction(cleanPrismaPromises)
+    }
+
+    describe('issue #8832', () => {
+      beforeEach(async () => {
+        await clean()
+      }, 10_000)
+
+      test('should succeed when "in" has 32766 or less ids', async () => {
+        const n = 32766
+        const ids = await createTags(n)
+        const tags = await getTagsFindManyIn(ids)
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "include" involves 32766 or less records', async () => {
+        const n = 32766
+        await createTags(n)
+        const tags = await getTagsFindManyInclude()
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "in" has 32767 or less ids', async () => {
+        const n = 32767
+        const ids = await createTags(n)
+        const tags = await getTagsFindManyIn(ids)
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "include" involves 32767 or less records', async () => {
+        const n = 32767
+        await createTags(n)
+        const tags = await getTagsFindManyInclude()
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "in" has 32768 or less ids', async () => {
+        const n = 32768
+        const ids = await createTags(n)
+        const tags = await getTagsFindManyIn(ids)
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "include" involves 32768 or less records', async () => {
+        const n = 32768
+        await createTags(n)
+        const tags = await getTagsFindManyInclude()
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "in" has 65536 or less ids', async () => {
+        const n = 65536
+        const ids = await createTags(n)
+        const tags = await getTagsFindManyIn(ids)
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "include" involves 65536 or less records', async () => {
+        const n = 65536
+        await createTags(n)
+        const tags = await getTagsFindManyInclude()
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should fail with `value too large to transmit` when "in" is repeated at least twice and "n" is $QUERY_BATCH_SIZE', async () => {
```

Comment:
This and following test cases (except the last one) are the ones we did not fix yet, correct?
So when we do fix that, these tests here would also need to pass?
Do we have an issue for that work yet? 

I think it would make sense to group this into its own "describe" and link to the issue for context.